Simplify get() now that we don't need it to ignorespaces

edemaine opened the pull request:
Continuation of #912 (forgot to do this change there before merging).

Open question: Should we keep the `get()` alias for `expandNextToken()`, or just use the latter?  `unget()` seems like a useful thing to keep (though it's no longer used), but we could also rename it (e.g. `pushToken`).

kevinbarabash commented on src/Parser.js:
```diff
@@ -140,17 +140,18 @@ export default class Parser {
+        //this.gullet.unget(this.nextToken);
         this.mode = newMode;
-        this.consume();
+        //this.consume();
```
remove commented out code

kevinbarabash replied:
I'm curious though why the `unget` and `consume` are no longer necessary.

edemaine replied:
It used to be that `get` got passed true or false according to whether math mode was on, i.e., it ate spaces if it was in math mode.  But that's no longer the case (having moved that into parser, where it belongs according to TeX behavior), so now get doesn't depend on whether math mode is on, so there's no effect to ungetting and regetting.

kevinbarabash commented on src/Parser.js:
```diff
@@ -140,17 +140,18 @@ export default class Parser {
-     * in case it would be read differently in the new mode.
+     * Switches between "text" and "math" modes.
+     * This used to reconsume nextToken in the new mode because whitespace
+     * handling differed between the two nodes; now it just updates this.mode.
```
Describing the previous behavior isn't necessary.

edemaine commented:
@kevinbarabash I've removed the excess documentation and commented-out old behavior.

kevinbarabash commented:
I'll finish reviewing this evening.

kevinbarabash commented on src/MacroExpander.js:
```diff
@@ -227,36 +225,17 @@ export default class MacroExpander implements MacroContextInterface {
-     * was got in the old mode but should get got again in a new mode
-     * with possibly different whitespace handling.
      */
     unget(token: Token) {
```
Since it's not being used and it's easy enough to do `this.stack.push(token)`, let's get rid of this.

edemaine replied:
I think it's still a nice interface to keep, in case we want to use it in macros.js. Though still happy to rename it...

kevinbarabash replied:
If you rename it, please change all `this.stack.push(token)`s to use it.

edemaine commented:
I dropped the `get()` alias (opting for `expandNextToken` instead), renamed `unget()` to `pushToken()`, and used it in two places as suggested by @kevinbarabash (good catch!).  There is one exception: I couldn't easily change `this.stack.push(...expansion)` given the different argument type.  Should I change `pushToken()` to use `arguments`...?  There's also a new doc string for `pushToken` that'd be worth reading.

kevinbarabash commented:
I think it's fine for `pushToken` to only accept a single token for now.  If we find we're needing to push multiple tokens at once in more places in the future we can revisit at that time.

kevinbarabash commented:
@edemaine thanks the PR.
